src/mpyl/steps/deploy/k8s/__init__.py

```diff
@@ -57,6 +60,22 @@ def upsert_namespace(
         logger.info(f"Found namespace {namespace}")
 
 
+def delete_namespace_if_exists(
+    logger: Logger, namespace: str, rancher_config: ClusterConfig
+):
```

Comment:
I think we should build in some safety features for this function to only delete for the test cluster and only namespaces that include `pr`

Reply:
It does only delete in the test cluster

Reply:
I'll add a check for namespaces named `pr-XXXX`

Reply:
I know you pass the test cluster config here, but it's completely possible to pass the config for production right?

Reply:
ah you mean the method itself

Reply:
it's not easy since there's no guarantees what the test clusters name is.. And in the end it's just a ClusterConfig

Reply:
You can check for `clusterEnv`?

Reply:
fair